app/controllers/sessions_controller.rb

```diff
@@ -37,8 +42,12 @@ def destroy
     # that are not at the root of their host after logout
     # TODO: Replace with signed or registered return urls
     #       Need to provide web views to sign or register those urls
-    uri = URI(request.referer)
-    url = "#{uri.scheme}://#{uri.host}:#{uri.port}/"
+    url = begin
+      uri = URI(request.referer)
+      "#{uri.scheme}://#{uri.host}:#{uri.port}/"
+    rescue # in case the referer is bad (see #179)
+      root_url
+    end
```

Comment:
But of course that means you need to calculate this before you clear the client_id